.github/workflows/ci.yml

```diff
@@ -0,0 +1,63 @@
+name: Continuos Integration
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+  schedule:
+    - cron:  '0 0 * * 1' # Every Monday at 00:00 UTC
+
+jobs:
+  golang-check:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        go-version: [
+          "1.21",
+        ]
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v4
+        with:
+            go-version: ${{ matrix.go-version }}
+
+      - name: Install Go dependencies
+        run: go mod download
+
+      - name: Go unit and integration tests
+        run: make simulation-test
+
+    # TODO coverage report
+
+      - name: Go linters
+        run: make lint
+
+  container-check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Build container image
```

Comment:
there is no need to build the image, added just to check what you think. Should we keep?

Reply:
We should build the container and host it in quay I think.

I have added the following three github secrets:
```
QUAY_USERNAME
QUAY_PASSWORD
QUAY_IMAGE_NAMESPACE
```

To be uses same as:
   https://github.com/dora-metrics/pelorus/blob/master/.github/workflows/cd.yml#L79-L81

Which is using template:
   https://github.com/dora-metrics/pelorus/blob/master/.github/workflow_templates/build_push_exporters/action.yml

Reply:
@mateusoliveira43 I think we need to build it as the oadp manager will deploy that controller from the image itself, so it has to be build and hosted somewhere. I think the best now is to use latest during development.